components/scream/src/physics/p3/p3_functions_f90.hpp

```diff
@@ -479,34 +480,34 @@ void get_rain_dsd2_f(Real qr, Real* nr, Real* mu_r, Real* lamr, Real* cdistr, Re
 
 ///////////////////////////////////////////////////////////////////////////////
 
-struct CalcUpwindData
+struct CalcUpwindData : public PhysicsTestData
 {
   // Inputs
   Int kts, kte, kdir, kbot, k_qxtop, num_arrays;
   Real dt_sub;
   Real* rho, *inv_rho, *inv_dz;
-  Real **vs;
+  Real *vs; // num_arrays x nk
 
   // In/out
-  Real **qnx;
+  Real *qnx; // num_arrays x nk
 
   // Outputs
-  Real** fluxes;
+  Real *fluxes; // num_arrays x nk
 
-  CalcUpwindData(Int kts_, Int kte_, Int kdir_, Int kbot_, Int k_qxtop_, Int num_arrays_, Real dt_sub_,
-                 std::pair<Real, Real> rho_range, std::pair<Real, Real> inv_dz_range,
-                 std::pair<Real, Real> vs_range, std::pair<Real, Real> qnx_range);
+  CalcUpwindData(Int kts_, Int kte_, Int kdir_, Int kbot_, Int k_qxtop_, Int num_arrays_, Real dt_sub_);
 
   // deep copy
   CalcUpwindData(const CalcUpwindData& rhs);
 
+  CalcUpwindData& operator=(const CalcUpwindData& rhs) = delete;
```

Comment:
Is it possible to delete the assignment operator in the base class and have subclasses follow suit? Is there any reason we wouldn't want to do this?